### ISSUE TITLE
Right Click -> "X Axis" -> "Auto" generates "TypeError: setValue(self, int): argument 1 has unexpected type 'float'"

### ISSUE THREAD

**Issue by MicheleCotrufo** [pyqtgraph/pyqtgraph#2196]
### Short description
I generated a test plot (see code below). When I right-click on it, it opens the context menu. I click on 'X axis'. The modality 'auto' is already selected. If I click on 'manual' and then back on 'auto, it generates the following error

```
Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\pyqtgraph\graphicsItems\ViewBox\ViewBoxMenu.py", line 102, in viewStateChanged
    self.updateState()
  File "C:\Python310\lib\site-packages\pyqtgraph\graphicsItems\ViewBox\ViewBoxMenu.py", line 120, in updateState
    self.ctrl[i].autoPercentSpin.setValue(state['autoRange'][i]*100)
TypeError: setValue(self, int): argument 1 has unexpected type 'float'
```

### Code to reproduce

```python
from PyQt5 import QtWidgets
from pyqtgraph import PlotWidget, plot
import pyqtgraph as pg
import sys  
import os

class MainWindow(QtWidgets.QMainWindow):

    def __init__(self, *args, **kwargs):
        super(MainWindow, self).__init__(*args, **kwargs)

        self.graphWidget = pg.PlotWidget()
        self.setCentralWidget(self.graphWidget)

        hour = [1,2,3,4,5,6,7,8,9,10]
        temperature = [30,32,34,32,33,31,29,32,35,45]

        # plot data: x, y values
        self.graphWidget.plot(hour, temperature)

def main():
    app = QtWidgets.QApplication(sys.argv)
    main = MainWindow()
    main.show()
    sys.exit(app.exec_())

if __name__ == '__main__':
    main()
```

### Tested environment(s)

 * PyQtGraph version: I tested the versions 0.12.0, 0.12.1, 0.12.2 and 0.12.3 and they all create this error
 * Qt Python binding: 5.15.6
 * Python version:  3.10.2
 * NumPy version: 1.22.2
 * Operating system: Windows 10
 * Installation method: pip

**Comment by MicheleCotrufo** [comment]
Thanks, it worked!
Do you know when will this fix be implemented in a release?

**Comment by j9ac9k** [comment]
we have a number of critical bug-fixes ready for the next release; there is
another new-feature PR Im' hoping to get merged and then issue the next
release; which for a series of circumstances have delayed me for some time.

tl;dr it will be soon (1-2 weeks?), I'm starting to entertain the idea of
releasing 0.12.4 even if that other PR doesn't get in.

**Comment by j9ac9k** [comment]
Fixed in #2153
